### PR TITLE
Add integrations product. Create Client resource.

### PR DESCRIPTION
**Pull request by rohitjangid**
Add support to manage Application Integration Client via terraform

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_integrations_client
```

Fixes https://github.com/hashicorp/terraform-provider-google/issues/17567

### CODE REVIEW

File: mmv1/products/integrations/product.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+name: Integrations
+display_name: Application Integration
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://integrations.googleapis.com/v1/
+  # Integrations has no beta API, but running the full Terraform
+  # acceptance tests depends on a beta-only resource and a bug in MM
+  # will generate beta tests against the GA provider in this case, so
+  # we fake a beta API with the same endpoint as the GA API.
```
Comment (c2thorn) [inline mmv1/products/integrations/product.yaml:24]: Is this confirmed by yourself or copied from somewhere else

**Comment by c2thorn** [comment]
please add tests for the fields pointed out in https://github.com/GoogleCloudPlatform/magic-modules/pull/10186#issuecomment-1995157201

I've enabled the API in our test project, so we won't get that failure with the next push.

**Comment by rohitjangid** [comment]
@c2thorn I have done the requested changes.

**Comment by rohitjangid** [comment]
Hi @c2thorn 
I am not sure teamcity-services-diff-check test is failing. Can you help me fix this?

**Comment by c2thorn** [comment]
you'll need to add the new service to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt

@rohitjangid

**Comment by rohitjangid** [comment]
> you'll need to add the new service to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt
> 
> @rohitjangid

@c2thorn I added the service to this file but the test is still failing

**Comment by c2thorn** [comment]
> > you'll need to add the new service to [`main`/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt?rgh-link-date=2024-03-15T17%3A01%3A37Z)
> > @rohitjangid
> 
> @c2thorn I added the service to this file but the test is still failing

apologies, looks like there is one more file `services_beta.kt`
this is a relatively new requirement I did not know about, sorry for the inconvenience. I've looked and cannot find a third

**Comment by rohitjangid** [comment]
> > > you'll need to add the new service to [`main`/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt?rgh-link-date=2024-03-15T17%3A01%3A37Z)
> > > @rohitjangid
> > 
> > 
> > @c2thorn I added the service to this file but the test is still failing
> 
> apologies, looks like there is one more file `services_beta.kt` this is a relatively new requirement I did not know about, sorry for the inconvenience. I've looked and cannot find a third

@c2thorn Thanks. It seems to be passing now. These files are missing from https://googlecloudplatform.github.io/magic-modules/get-started/contribution-process/ which is causing this issue.

File: .ci/infra/terraform/main.tf [hunk 2]
```diff
@@ -259,6 +259,7 @@ module "project-services" {
     "iap.googleapis.com",
     "identitytoolkit.googleapis.com",
     "ids.googleapis.com",
+    "integrations.googleapis.com",
```
Comment (c2thorn) [inline .ci/infra/terraform/main.tf:262]: apologies again, we need to remove this for now. I can add it in a separate PR, it's not necessary to add at this moment.

Comment (rohitjangid) [reply .ci/infra/terraform/main.tf]: Done

**Comment by rohitjangid** [comment]
> Tests passed during RECORDING mode: `TestAccIntegrationsClient_integrationsClientAdvanceExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10186/artifacts/f4ec31ad-546e-4ec2-8c7e-6b23442b2514/recording/TestAccIntegrationsClient_integrationsClientAdvanceExample.log)]
> 
> Tests failed when rerunning REPLAYING mode: `TestAccIntegrationsClient_integrationsClientAdvanceExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10186/artifacts/f4ec31ad-546e-4ec2-8c7e-6b23442b2514/build-log/replaying_build_after_recording/TestAccIntegrationsClient_integrationsClientAdvanceExample_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10186/artifacts/f4ec31ad-546e-4ec2-8c7e-6b23442b2514/replaying_after_recording/TestAccIntegrationsClient_integrationsClientAdvanceExample.log)]
> 
> Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.
> 
> Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.
> 
> All tests passed! View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10186/artifacts/f4ec31ad-546e-4ec2-8c7e-6b23442b2514/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10186/artifacts/f4ec31ad-546e-4ec2-8c7e-6b23442b2514/recording) for each test

The error is from kms api. I am not sure what should be the right way to fix this. @c2thorn Do you have any suggestions for the same?

File: mmv1/templates/terraform/examples/integrations_client_advance.tf.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+data "google_project" "test_project" {
+}
+
+resource "google_kms_key_ring" "keyring" {
+  name = "tf-test-keyring"
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/integrations_client_advance.tf.erb:5]: let's try making the key ring and key names variables like here: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/data_fusion_instance_cmek.tf.erb#L14

this way we can avoid collision errors.

Comment (rohitjangid) [reply mmv1/templates/terraform/examples/integrations_client_advance.tf.erb]: Done. Added vars and referred that var here.

**Comment by c2thorn** [comment]
@rohitjangid It passes, but does not like our replay system. I will make a comment to show you how to remove it from our VCR steps and we will continue.

File: mmv1/products/integrations/Client.yaml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "integrations_client_basic"
+    primary_resource_id: "example"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "integrations_client_advance"
+    primary_resource_id: "example"
+    vars:
```
Comment (c2thorn) [inline mmv1/products/integrations/Client.yaml:110]:
```suggestion
    skip_vcr: true
    vars:
```

**Comment by c2thorn** [comment]
committing it myself to move things along.

**Review by c2thorn — APPROVED** [review]
LGTM, thanks for your patience.

**Pull request merged**